tests/test_helper/microovn.bash

```diff
@@ -637,23 +637,36 @@ function microovn_lsp_up() {
     test "$result" -eq 1
 }
 
-# microovn_add_vif CONTAINER NSNAME IFNAME
+# microovn_add_vif CONTAINER NS_NAME IF_NAME [LS_NAME [LSP_IP]]
 #
-# Create LSP in LS for CONTAINER, create OVS internal interface with IFNAME,
-# attach it to LSP and move it into network namespace NSNAME.
+# Create LSP in LS for CONTAINER, create OVS internal interface with IF_NAME,
+# attach it to LSP and move it into network namespace NS_NAME.
 #
-# LLADDR and CIDR is generated based on integer found after the last ``-`` in
-# CONTAINER which currently needs to be a value between 0-9.
+# LS name is either automatically generated based on CONTAINER name and
+# MICROOVN_PREFIX_LS, or it can be optionally specified with LS_NAME argument.
+#
+# IP address for the LSP/VIF can be optionally provided as LSP_IP argument (in
+# CIDR format). Otherwise it will be automatically generated (along with MAC
+# address) based on integer found after the last ``-`` in CONTAINER which
+# currently needs to be a value between 0-9.
 function microovn_add_vif() {
     local container=$1; shift
     local ns_name=$1; shift
     local if_name=$1; shift
+    local ls_name=$1; shift
+    local cidr=$1; shift
+
+    if [ -z "$ls_name" ]; then
+        ls_name="${MICROOVN_PREFIX_LS}-${container}"
+    fi
+
+    if [ -z "$cidr" ]; then
+        cidr="10.42.$n.10/24"
```

Comment:
This is called prior to `n` being defined, however I see this code has already diverged from main, so instead of having you do another iteration on this bit I forwarded your patch with some modifications to a PR for main: canonical/microovn#204

In that patch I also fix the use of `shift` above, which in its current form does not make the options optional.

Reply:
Thanks for opening that "backport" to `main` and the fixes. I tested the `shift` behavior without `set -e` so I didn't noticed that it returns non-zero RC if it runs out of arguments.